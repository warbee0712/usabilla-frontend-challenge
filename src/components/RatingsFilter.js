import React from 'react'
import PropTypes from 'prop-types'
// Components
import Row from './Row'
import RoundButton from './RoundButton'

function RatingsFilter({ options, active, toggleFilter }) {
  return (
    <Row gutter="s">
      {options.map(item => (
        <RoundButton
          key={item}
          onClick={() => toggleFilter(item)}
          active={!!active.find(el => el === item)}
        >
          {item}
        </RoundButton>
      ))}
    </Row>
  )
}

RatingsFilter.propTypes = {
  options: PropTypes.arrayOf(PropTypes.number).isRequired,
  active: PropTypes.arrayOf(PropTypes.number).isRequired,
  toggleFilter: PropTypes.func.isRequired,
}

export default RatingsFilter
