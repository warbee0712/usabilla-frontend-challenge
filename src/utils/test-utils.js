import React from 'react'
import { render } from 'react-testing-library'
import { ThemeProvider } from 'styled-components'
import theme from '../theme'

const customRender = (node, options) => {
  return render(<ThemeProvider theme={theme}>{node}</ThemeProvider>, options)
}

// re-export everything
export * from 'react-testing-library'

// override render method
export { customRender as render }
