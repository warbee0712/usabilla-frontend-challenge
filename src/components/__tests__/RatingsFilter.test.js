import React from 'react'
import { render } from 'test-utils'
import RatingsFilter from '../RatingsFilter'
import { fireEvent } from 'react-testing-library'

describe('RatingsFilter', () => {
  it('should match snapshot', () => {
    const { container } = render(
      <RatingsFilter
        options={[1, 2, 3, 4, 5]}
        active={[1, 2, 3]}
        toggleFilter={() => {}}
      />
    )
    const node = container.firstChild

    expect(node).toMatchSnapshot()
  })
  it('should render all options', () => {
    const { getAllByText } = render(
      <RatingsFilter
        options={[1, 2, 3, 4, 5]}
        active={[1, 2, 3]}
        toggleFilter={() => {}}
      />
    )

    const buttons = getAllByText(/[0-9]/)

    expect(buttons).toHaveLength(5)
    // test that all options have the right text
    expect(buttons[0].textContent).toBe('1')
    expect(buttons[1].textContent).toBe('2')
    expect(buttons[2].textContent).toBe('3')
    expect(buttons[3].textContent).toBe('4')
    expect(buttons[4].textContent).toBe('5')
    // test that 1,2,3 are active and 4,5 are not
    expect(buttons[0].className).toEqual(buttons[1].className)
    expect(buttons[1].className).toEqual(buttons[2].className)
    expect(buttons[2].className).not.toEqual(buttons[3].className)
    expect(buttons[3].className).toEqual(buttons[4].className)
  })
  it('should fire toggle function when a filter button is clicked', () => {
    const handleToggle = jest.fn()
    const { getByText } = render(
      <RatingsFilter
        options={[1, 2, 3, 4, 5]}
        active={[1, 2, 3]}
        toggleFilter={handleToggle}
      />
    )

    const FilterButton = getByText('1')
    fireEvent.click(FilterButton)

    expect(handleToggle).toHaveBeenCalledTimes(1)
    expect(handleToggle).toHaveBeenCalledWith(1)
  })
})
