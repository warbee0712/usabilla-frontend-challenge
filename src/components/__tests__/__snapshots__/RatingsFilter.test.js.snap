// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RatingsFilter should match snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c0 > * + * {
  margin-left: 0.6875rem;
}

.c1 {
  background-color: transparent;
  color: #59636b;
  border: 1px solid #59636b;
  border-radius: 9999em;
  -webkit-transition: color 0.2s cubic-bezier(0.38,1.59,0.82,0.96), box-shadow 0.2s cubic-bezier(0.38,1.59,0.82,0.96);
  transition: color 0.2s cubic-bezier(0.38,1.59,0.82,0.96), box-shadow 0.2s cubic-bezier(0.38,1.59,0.82,0.96);
  position: relative;
  cursor: pointer;
  outline: none;
  background-color: #00a5c9;
  border-color: #00a5c9;
  color: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 30px;
  width: 30px;
}

.c1:hover {
  background-color: #59636b;
  border-color: #59636b;
  box-shadow: 0 1px 2px rgba(0,0,0,0.15);
  color: #fff;
}

.c1:focus {
  border: 1px solid #00a5c9;
  box-shadow: 0 1px 2px rgba(0,0,0,0.15);
  outline: none;
}

.c1:hover {
  background-color: #008aab;
  border-color: #008aab;
  box-shadow: 0 1px 2px rgba(0,0,0,0.15);
  color: #fff;
}

.c2 {
  background-color: transparent;
  color: #59636b;
  border: 1px solid #59636b;
  border-radius: 9999em;
  -webkit-transition: color 0.2s cubic-bezier(0.38,1.59,0.82,0.96), box-shadow 0.2s cubic-bezier(0.38,1.59,0.82,0.96);
  transition: color 0.2s cubic-bezier(0.38,1.59,0.82,0.96), box-shadow 0.2s cubic-bezier(0.38,1.59,0.82,0.96);
  position: relative;
  cursor: pointer;
  outline: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 30px;
  width: 30px;
}

.c2:hover {
  background-color: #59636b;
  border-color: #59636b;
  box-shadow: 0 1px 2px rgba(0,0,0,0.15);
  color: #fff;
}

.c2:focus {
  border: 1px solid #00a5c9;
  box-shadow: 0 1px 2px rgba(0,0,0,0.15);
  outline: none;
}

@media screen and (min-width:40em) {
  .c1 {
    height: 40px;
    width: 40px;
  }
}

@media screen and (min-width:40em) {
  .c2 {
    height: 40px;
    width: 40px;
  }
}

<div
  class="c0"
  display="flex"
>
  <button
    class="c1"
  >
    1
  </button>
  <button
    class="c1"
  >
    2
  </button>
  <button
    class="c1"
  >
    3
  </button>
  <button
    class="c2"
  >
    4
  </button>
  <button
    class="c2"
  >
    5
  </button>
</div>
`;
